### PR TITLE
Deprecation and alternatives for utf8_encode and utf8_decode

### PR DESCRIPTION
**Pull request by IMSoP**

### CODE REVIEW

File: reference/strings/functions/utf8-decode.xml [hunk 1]
```diff
@@ -134,6 +134,107 @@ string(1) "?"
...
+5a6feb
+5a6feb
+5a6feb
+]]>
+     </screen>
+    </informalexample>
+    Specifying <literal>'?'</literal> as the <literal>'to_subst'</literal> option
+    to <methodname>UConverter::transcode</methodname> gives the same result as
+    <function>utf8_decode</function> for strings which are invalid or which can not
+    be represented in ISO 8859-1
```
Comment (cmb69) [inline reference/strings/functions/utf8-decode.xml:215]: Missing period:
```suggestion
    be represented in ISO 8859-1.
```

File: reference/strings/functions/utf8-decode.xml [hunk 2]
```diff
@@ -134,6 +134,107 @@ string(1) "?"
...
+    <informalexample>
+     <programlisting role="php">
+<![CDATA[
+$utf8_string = "\xE2\x82\xAC"; // € (Euro Sign) does not exist in ISO 8859-1
```
Comment (cmb69) [inline reference/strings/functions/utf8-decode.xml:219]: Missing `<?php` (which prevents syntax highlighting):
```suggestion
<?php
$utf8_string = "\xE2\x82\xAC"; // € (Euro Sign) does not exist in ISO 8859-1
```

File: reference/strings/functions/utf8-decode.xml [hunk 3]
```diff
@@ -134,6 +134,107 @@ string(1) "?"
   </example>
  </refsect1>
 
+ <refsect1 role="notes">
+  &reftitle.notes;
+  <note>
+   <title>Deprecation and alternatives</title>
+   <para>
+    This function has been <emphasis>deprecated</emphasis> as of PHP 8.2.0,
```
Comment (cmb69) [inline reference/strings/functions/utf8-decode.xml:142]: For a long time I wonder whether that should be "has been" or "is"; I think the latter is correct, but I may be wrong.

**Review by cmb69 — COMMENTED** [review]
Very nice, thank you!

File: reference/strings/functions/utf8-decode.xml (hunk 3 above)
Comment (IMSoP) [reply reference/strings/functions/utf8-decode.xml]: I agree, that reads better in this context. I've left it as "has been" in the changelog, though, to match "has been moved".

**Review by Girgias — APPROVED** [review]

**Pull request merged**